### PR TITLE
chore: add E2E tests for Cloud Run

### PR DESCRIPTION
**Pull request by averikitsch**
Fixes #issue

> It's a good idea to open an issue first for discussion.

- [ ] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] API's need to be enabled to test (tell us)
- [ ] Environment Variables need to be set (ask us to set them)
- [ ] **Tests** pass:   `mvn clean verify` **required**
- [ ] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] Please **merge** this PR for me once it is approved.

### CODE REVIEW

**Comment by averikitsch** [comment]
Project needs to be allowlisted

File: run/jobs/pom.xml [hunk 1]
```diff
@@ -46,6 +46,12 @@ limitations under the License.
       <version>1.1.3</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>com.google.cloud</groupId>
+      <artifactId>google-cloud-logging</artifactId>
+      <version>3.6.2</version>
+      <scope>test</scope>
```
Comment (lesv) [inline run/jobs/pom.xml:53]: Why is this just `test` scope?

File: run/jobs/src/test/java/com/example/JobsIntegrationTests.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+public class JobsIntegrationTests {
+
+  private static final String project = System.getenv("GOOGLE_CLOUD_PROJECT");
+  private static final String suffix = UUID.randomUUID().toString();
+  private static String service;
+
+  @BeforeClass
+  public static void setup() throws Exception {
+    service = "job-quickstart-f374953a-498c-43b8-847a-7998ae5665d5"; // "job-quickstart-" + suffix;
```
Comment (lesv) [inline run/jobs/src/test/java/com/example/JobsIntegrationTests.java:49]: Is this really a fixed thing?

Comment (lesv) [reply run/jobs/src/test/java/com/example/JobsIntegrationTests.java:49]: looking at the code, perhaps this should be a real UUID?

File: run/jobs/src/test/java/com/example/JobsIntegrationTests.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+    ProcessBuilder setup = new ProcessBuilder();
+    setup.command(
+        "gcloud",
+        "builds",
+        "submit",
+        "--project=" + project,
+        "--config=./src/test/java/com/example/resources/e2e_test_setup.yaml",
+        String.format("--substitutions=_SERVICE=%s,_VERSION=%s", service, suffix));
```
Comment (lesv) [inline run/jobs/src/test/java/com/example/JobsIntegrationTests.java:58]: I always wonder why we do this?  Why we think it's a good idea?

File: run/jobs/src/test/java/com/example/resources/e2e_test_cleanup.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+steps:
```
Comment (lesv) [inline run/jobs/src/test/java/com/example/resources/e2e_test_cleanup.yaml:1]: You could have a copyright header here.

File: run/jobs/src/test/java/com/example/resources/e2e_test_setup.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
+steps:
```
Comment (lesv) [inline run/jobs/src/test/java/com/example/resources/e2e_test_setup.yaml:1]: You could also have a copyright header here

**Review by lesv — APPROVED** [review]
* Please answer questions
* Add copyrights on human generated content.
* I'm not a fan of code that executes commands... Are you sure you wouldn't be better off w/ a script?
* Think about what happens if this is executed in 4 VM's at the same time.  Would it work?  Why / Why not?  (I'm talking about creating builders with the same name, then deleting them)

File: run/jobs/pom.xml (hunk 1 above)
Comment (averikitsch) [reply run/jobs/pom.xml:53]: The tests are looking for the logs in Cloud Logging. The sample writes to stdout.

File: run/jobs/src/test/java/com/example/JobsIntegrationTests.java (hunk 3 above)
Comment (averikitsch) [reply run/jobs/src/test/java/com/example/JobsIntegrationTests.java:58]: What specifically? Using substitutions - I want my job and image to have UUID tags, using String.format - looks nicer than +, Cloud Build - no "easy" programmatic way to build a container or deploy to Cloud Run.

Comment (lesv) [reply run/jobs/src/test/java/com/example/JobsIntegrationTests.java:58]: Running commands from code in our tests.

**Review by lesv — APPROVED** [review]

**Review by lesv — APPROVED** [review]

**Pull request merged**